### PR TITLE
Fix for php 8.0

### PR DESCRIPTION
**Pull request by pandarek**
$parts must be an array, because the argument for count() must be countable, mb_split() returns an array or a boolean.

### CODE REVIEW

File: libs/plugins/shared.mb_str_replace.php [hunk 1]
```diff
@@ -44,7 +44,8 @@ function smarty_mb_str_replace($search, $replace, $subject, &$count = 0)
                 }
             }
         } else {
-            $parts = mb_split(preg_quote($search), $subject);
+            $parts = mb_split(preg_quote($search, null), $subject);
```
Comment (wisskid) [inline libs/plugins/shared.mb_str_replace.php:47]: `, null` seems unnecessary?

File: libs/plugins/shared.mb_str_replace.php [hunk 2]
```diff
@@ -44,7 +44,8 @@ function smarty_mb_str_replace($search, $replace, $subject, &$count = 0)
...
+            $parts = $parts ?: array();
```
Comment (wisskid) [inline libs/plugins/shared.mb_str_replace.php:48]: you could add `?: array()` to the end of the previous line as well

File: libs/plugins/shared.mb_str_replace.php (hunk 1 above)
Comment (pandarek) [reply libs/plugins/shared.mb_str_replace.php:47]: Correct, it can be omitted

File: libs/plugins/shared.mb_str_replace.php (hunk 2 above)
Comment (pandarek) [reply libs/plugins/shared.mb_str_replace.php:48]: yes, it will be nicer

**Comment by pandarek** [comment]
corrections made

**Review by wisskid — APPROVED** [review]

**Comment by wisskid** [comment]
@pandarek  thank you!

**Pull request merged**